src/components/ConditionSelector/Condition.tsx

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import './Condition.scss';
+import * as classNames from 'classnames';
+import { ConditionRowState, ConditionDefinition, PropertyTypeEnum } from './';
+import { Dropdown, DropdownType, IDropdownOption } from '../Dropdown';
+import { Button } from '../Button';
+import { TextField } from '../TextField';
+import { DateTimeDropdownPicker } from '../DateTimeDropdownPicker/DateTimeDropdownPicker';
+import { DragSource } from 'react-dnd';
+
+const conditionSource = {
+    beginDrag(props) {
+        return {};
+    },
+    endDrag(props, monitor, component) {
+        const dropTarget = monitor.getDropResult();
+        if (Boolean(props) && Boolean(dropTarget) && component.props.conditionDragged) {
+            component.props.conditionDragged(props, dropTarget);
+        }
+        return {};
+    }
+};
+
+function collectSource(connect, monitor) {
+    return {
+        connectDragSource: connect.dragSource(),
+        connectDragPreview: connect.dragPreview(),
+        isDragging: monitor.isDragging()
+    };
+}
+
+@DragSource('condition', conditionSource, collectSource)
+export class Condition extends React.PureComponent <ConditionDefinition, any> {
```

Comment:
It would be better to use an specific interface for state (or simply if {} if this component does not have state) instead of any. 
This way we leverage TypeScript features and nobody can mistakenly call this.setState.